packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,158 @@
+    const setLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(setLogRetention);
+    const stack = new cdk.Stack(app);
+
+    let websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      LoggingConfig: {
+        LogGroup: {
+          Ref: Match.stringLikeRegexp('CustomCDKBucketDeployment'),
+        },
+      },
+    });
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: setLogRetention,
+    });
+  });
+
+  test('when CustomResourceConfig.addLogRetentionLifetime is called with extra logGroup, extra logGroup remain unmodified by CustomResourceConfig', () => {
+    const setLogRetention = logs.RetentionDays.TEN_YEARS;
+    const default_log_group_retention = logs.RetentionDays.TWO_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(setLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const ignored = new logs.LogGroup(stack, 'ignored', {});
+    let websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.resourceCountIs('AWS::Logs::LogGroup', 2);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: setLogRetention,
+    });
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: default_log_group_retention,
+    });
+  });
+});
+
+describe('when logRetention is specified', () => {
+  test('when CustomResourceConfig.addLogRetentionLifetime is called, singleton-backed custom resource logRetention period modified', () => {
+    const setLogRetention = logs.RetentionDays.TEN_YEARS;
+    const locallySetLogRetention = logs.RetentionDays.ONE_WEEK;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(setLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployLogRetention', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logRetention: locallySetLogRetention,
+    });
+    const template = Template.fromStack(stack);
+
+    const captureRetentionInDays = new Capture();
+    template.hasResourceProperties('Custom::LogRetention', {
+      RetentionInDays: captureRetentionInDays,
```

Comment:
`Capture` is a cool feature but you don't need to use it here. It's enough to say:

```suggestion
      RetentionInDays: logRetentionOverride,
```